### PR TITLE
Add rcl_node_get_fully_qualified_name

### LINKED ISSUE #255: Get fully qualified node name
## Feature request

A fully qualified node name is the combination of the node's namespace and name. This is a request for a function to get that from a node.

#### Feature description
https://github.com/ros2/rclcpp/blob/ef427a7e3c1d8d8b1c7f04e01a4de318694618bb/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L93-L105

#### Implementation considerations
`char * rcl_node_get_fully_qualified_name(rcl_node_t *)` ?

### PR DESCRIPTION
**Pull request by rarvolt**
This fixes #255

### CODE REVIEW

File: rcl/include/rcl/node.h [hunk 1]
```diff
@@ -352,6 +352,29 @@ RCL_WARN_UNUSED
...
+/**
+ * This function returns the node's internal namespace and name combined string.
+ * This function can fail, and therefore return `NULL`, if:
+ *   - node is `NULL`
+ *   - node has not been initialized (the implementation is invalid)
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
```
[the remark this answers is not in the document]
Comment (rarvolt) [reply rcl/include/rcl/node.h]: Fixed.
`name` and `namespace` are stored in the specific rmw implementation (if I understand correctly) and I hesitated from making changes across many repos. Thats why I decided to store the computed `fq_name` in the `rcl_node_impl_t` structure.

**Comment by tfoote** [comment]
@rarvolt This looks good overall. It failed CI so I rebased it and pushed to a local branch. If you have a moment to rebase this yourself that would be great. As well we've added a DCO that would be great if you could sign off on by signing your commits. We're not yet strictly enforcing the DCO during the transition but it would be great if you'd be willing to try it out and give us any feedback at this stage. 

More info on the DCO is here: https://probot.github.io/apps/dco/

**Comment by tfoote** [comment]
The rebase passed in https://github.com/ros2/rcl/pull/381 If you have a chance to rebase it before tomorrow that would be great. Otherwise I'll merge my rebase.

**Comment by tfoote** [comment]
Superceded by #381

**Pull request closed without merging**